src/ubi.c

```diff
@@ -0,0 +1,148 @@
+/**
+ * collectd - src/ubi.c
+ *
+ * This program is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License as published by the
+ * Free Software Foundation; only version 2 of the License is applicable.
+ *
+ * This program is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License along
+ * with this program; if not, write to the Free Software Foundation, Inc.,
+ * 51 Franklin St, Fifth Floor, Boston, MA  02110-1301 USA
+ */
+
+#include "collectd.h"
+
+#include "plugin.h"
+#include "utils/common/common.h"
+#include "utils/ignorelist/ignorelist.h"
+
+#if !KERNEL_LINUX
+#error "No applicable input method."
+#endif
+
+#define PLUGIN_NAME "ubi"
+
+#define SYS_PATH "/sys/class/ubi/"
+
+/*
+ * Device attributes
+ */
+#define DEV_BAD_COUNT                                                          \
+  "bad_peb_count" // Count of bad physical eraseblocks on the underlying MTD
+                  // device.
+#define MAXIMUM_ERASE "max_ec" // Current maximum erase counter value
```

Comment:
Did this counter changes frequently or rarely?
Do you have sysfs documentation for all possible counters and their values for UBI/MTD?

From my understanding, this counter is very static. Is I'm correct?

Reply:
I think, `reserved_for_bad` is also very useful for your plugin, because it provides us number of eraseblocks reserved for bad block handling. If it becomes zero, there will be troubles when the next bad block appears. So plugin might watch this value too.

Reply:
According to the documentation:
_bad_peb_count_: count of bad physical eraseblocks
_max_ec_: current highest erase counter value

Those 2 counters *should* not move a lot and that's why it is good idea to monitor them.

Indeed, when the '_bad_peb_count_' starts to increase, something wrong is probably happening because your device is potentially dying.

Concerning '_max_ec_', this counter will increase by sure but it should not increase too fast, otherwise, it means you are using your flash too fast.

Reply:
The counter 'reserved_for_bad' will not move, it is static. There is no need to monitor it